### ISSUE TITLE
weird physics bug

### ISSUE THREAD

**Issue by yonimal676** [yonimal676/Basketball#46]
![image](https://user-images.githubusercontent.com/96010721/218088005-6504cce7-69a1-4a17-b5e5-1283c86d5f32.png)

This happens when the ball is shot at the approximate angle of  -177  degrees

previously it looked like this:
![image](https://user-images.githubusercontent.com/96010721/218088900-bc77403e-43d4-40fe-89fe-82991aba790b.png)
with this formula (v0y is minus because of the inverted Y axis):
`        ball.vy = - ball.v0y - (ball.GRAVITY * time);
`

but now the code is:
```
        if (quarterOfLaunch == 1 || quarterOfLaunch == 2) // throwing the ball downwards.
            ball.vy = ball.v0y - (ball.GRAVITY * time);  // the vertical velocity changes constantly.
        else 
            ball.vy = ball.v0y + (ball.GRAVITY * time);
```

After playing with the physics for a bit, this is the current state:

![image](https://user-images.githubusercontent.com/96010721/220990478-40b20b5d-b1ae-4d8a-bc2c-2386f1b14bb2.png)

## Basically what happens when the ball changes its' trajectory is that the VY reaches 0 after being minus (-1) and then flips to positive (1) as it should, but the velocity-changing system flips the number when it reaches max height.
Here's the solotion:
![image](https://user-images.githubusercontent.com/96010721/221128651-8304b1d6-85e5-4c3b-97c9-554da36b472b.png)
